### PR TITLE
Use Option<IpldBlock> for all message params

### PR DESCRIPTION
**Pull request by arajasek**
This PR:
- Updates FVM SDK and ipld_encoding
- Refactors to start using `Option<IpldBlock>` for message parameters

### CODE REVIEW

File: fvm_actor_utils/src/receiver/mod.rs [hunk 1]
```diff
@@ -119,10 +121,17 @@ impl<T: RecipientData> ReceiverHook<T> {
...
             RECEIVER_HOOK_METHOD_NUM,
-            &RawBytes::serialize(&params)?,
+            Some(IpldBlock::serialize(IPLD_RAW, &params_ser).map_err(|e| {
+                ReceiverHookError::IpldEncoding(fvm_ipld_encoding::Error {
+                    description: e.to_string(),
+                    protocol: fvm_ipld_encoding::CodecProtocol::Cbor,
+                })
+            })?),
             &TokenAmount::zero(),
         )?;
```
Comment (Stebalien) [inline fvm_actor_utils/src/receiver/mod.rs:136]: Why double-serialize?

Comment (Stebalien) [reply fvm_actor_utils/src/receiver/mod.rs:136]: Ah, I think this should be `Some(IpldBlock::serialize_cbor(params)?)`.

**Comment by arajasek** [comment]
CI failures are unrelated, see: https://github.com/helix-onchain/filecoin/pull/170

**Comment by anorth** [comment]
This library hasn't updated to fvm 3 yet. I'm interpreting this PR as sort of a prototype to check it will be ok to make this change, given the github dependencies etc.

**Comment by arajasek** [comment]
@anorth This PR won't be updating to FVM 3 (that will have to happen, but later). The GH dependencies will be removed, of course, but I would appreciate a review of the direction. I don't want to land [the corresponding FVM PR](https://github.com/filecoin-project/ref-fvm/pull/1199) without some confidence that the downstream consumers will be able to update.

File: fvm_actor_utils/src/messaging.rs [hunk 2]
```diff
@@ -152,7 +152,7 @@ impl Messaging for FvmMessenger {
 ///
 #[derive(Debug)]
 pub struct FakeMessenger {
-    pub last_message: RefCell<Option<RawBytes>>,
+    pub last_message: RefCell<Option<Option<IpldBlock>>>,
```
Comment (anorth) [inline fvm_actor_utils/src/messaging.rs:155]: Do we really need `Option<Option<>>`? This at least needs documentation to explain it.

**Review by anorth — COMMENTED** [review]
This looks fine directionally. I would approve when it's actually ready to land.

Can we get a helper method on IpldBlock, or just change `serialize_cbor` to return the `Some()` to reduce boilerplate here?

An alternative to evaluate for an FVM type to encapsulate `Option<IpldBlock>` so that most users don't have to know about this implementation detail, and avoid yet more boilterplate. These are both suggestions for the FVM rather than here.

**Comment by alexytsu** [comment]
These changes look reasonable to me

the check-build CI error is a simple linting error and can be suppressed by cherry-picking bafec14d2cd0cc4bb1f4249ac31868954b2340f0 or rebasing of the latest `main` commit

the linking error when building actors is complaning about some missing `opencl` library. I can solve this locally by installing `ocl-icd-opencl-dev` on Ubuntu but I'm not sure what's changed that requires this to build the actors

**Review by anorth — APPROVED** [review]

**Pull request merged**